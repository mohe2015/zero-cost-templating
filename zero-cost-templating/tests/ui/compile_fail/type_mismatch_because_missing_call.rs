#![feature(async_closure, async_iterator, coroutines, gen_blocks, noop_waker)]

extern crate alloc;

use alloc::borrow::Cow;

use zero_cost_templating::yields;
use zero_cost_templating_macros::template_stream;

#[template_stream("test.html.hbs")]
pub async gen fn test() -> Cow<'static, str> {
    let template = yields!(test_initial0());
    //let template = template.template0();
    let page_title = Cow::from("thetitle");
    let template = yields!(template.test_page_title1(page_title));
    let template = yields!(template.test_template2());
    let csrf_token = Cow::from("thetoken");
    let template = yields!(template.test_csrf_token3(csrf_token));
    let template = yields!(template.test_template4());
    let template = yields!(template.test_template10());
    let template = yields!(template.test_copyright_year11(Cow::from("2023".to_string())));
    yields!(template.test_template12());
}

pub fn main() {}
