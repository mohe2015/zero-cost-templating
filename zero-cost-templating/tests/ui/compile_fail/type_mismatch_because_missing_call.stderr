error[E0599]: no method named `test_page_title1` found for struct `Template<TestTemplate0, (), ()>` in the current scope
  --> tests/ui/compile_fail/type_mismatch_because_missing_call.rs:14:29
   |
9  | #[template_stream("test.html.hbs")]
   | ----------------------------------- method `test_page_title1` not found for this struct
...
14 |     let template = template.test_page_title1(page_title);
   |                             ^^^^^^^^^^^^^^^^ method not found in `Template<TestTemplate0, (), ()>`
   |
   = note: the method was found for
           - `Template<TestTemplate1, Partial, After>`
